stargate/crates/query-planner/src/context.rs

```diff
@@ -214,35 +217,35 @@ impl<'q> QueryPlanningContext<'q> {
         }
 
         let return_type = self.names_to_types.get(field_def.field_type.as_name());
-        let field_type_is_not_composite =
-            return_type.is_none() || !return_type.unwrap().is_composite_type();
 
-        if field_type_is_not_composite {
-            return vec![];
-        }
+        if let Some(return_type) = return_type {
+            if !return_type.is_composite_type() {
+                return vec![];
+            }
 
-        let return_type = return_type.unwrap();
+            let provided_fields = self
+                .get_key_fields(return_type, service_name, true)
+                .into_iter()
+                .map(|f| f.field_def.name);
 
-        let provided_fields = self
-            .get_key_fields(return_type, service_name, true)
-            .into_iter()
-            .map(|f| f.field_def.name);
-
-        if let Some(provides) = self.federation.provides(field_def) {
-            let fields = {
-                let mut visited_fragment_names: HashSet<&str> = HashSet::new();
-                let mut res = vec![];
-                collect_fields_names(
-                    provides,
-                    &self.fragments,
-                    &mut res,
-                    &mut visited_fragment_names,
-                );
-                res
-            };
-            provided_fields.chain(fields).collect()
+            if let Some(provides) = self.federation.provides(field_def) {
+                let fields = {
+                    let mut visited_fragment_names: HashSet<&str> = HashSet::new();
+                    let mut res = vec![];
+                    collect_fields_names(
+                        provides,
+                        &self.fragments,
+                        &mut res,
+                        &mut visited_fragment_names,
+                    );
+                    res
+                };
+                provided_fields.chain(fields).collect()
+            } else {
+                provided_fields.collect()
+            }
         } else {
-            provided_fields.collect()
+            vec![]
```

Comment:
is this potentially better as a default value?

Reply:
Tracking this in https://github.com/apollographql/federation/issues/477